subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/types.go

```diff
@@ -57,11 +58,49 @@ func (cl CellList) String() string {
 	return strings.Join(names, ", ")
 }
 
+func (cl CellList) removeCell(c Cell) CellList {
+	index := -1
+	for i, cc := range cl {
+		if cc == c {
```

Comment:
>if cc == c { [](start = 2, length = 12)

How does it compare object? By value or reference, deep compare or not?

Reply:
By reference. The type Cell is all pointers. I've tested. 

---
In reply to: [332499806](https://github.com/microsoft/pai/pull/3684#discussion_r332499806) [](ancestors = 332499806)

Reply:
This means that you need to be aware that cc and c does not make a copy to each other. Is it too strict?

---
In reply to: [334217427](https://github.com/microsoft/pai/pull/3684#discussion_r334217427) [](ancestors = 334217427,332499806)

Reply:
I mean, as long as cc and c point to the same object, they are considered equal. It doesn't matter if cc and c are copies or whatever.

---
In reply to: [334217590](https://github.com/microsoft/pai/pull/3684#discussion_r334217590) [](ancestors = 334217590,334217427,332499806)

Reply:
I mean the object is copied to be another different object. Pls comment the method that it is removed by reference.

---
In reply to: [334218030](https://github.com/microsoft/pai/pull/3684#discussion_r334218030) [](ancestors = 334218030,334217590,334217427,332499806)